cli/azd/internal/cmd/add/util.go

```diff
@@ -48,6 +48,25 @@ func validateResourceName(name string, prj *project.ProjectConfig) error {
 	return nil
 }
 
+// validateContainerName validates storage account container names.
+// Reference: https://learn.microsoft.com/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names
+func validateContainerName(name string) error {
+	if len(name) < 3 {
+		return errors.New("name must be 3 characters or more")
+	}
+
+	if strings.Contains(name, "--") {
+		return errors.New("name cannot contain consecutive hyphens")
+	}
+
+	err := names.ValidateLabelName(name)
```

Comment:
I noticed `ValidateLabelName` uses a regex that checks for a minimum string length of 2 - is that intentional? I couldn't find in the RFC 1123 spec that it had to have a min length of 2.

```
var rfc1123LabelRegex = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9]$`)
...
if !rfc1123LabelRegex.MatchString(name) {
		return errors.New("name must contain only lower-cased alphanumeric characters or '-', i.e. a-z, 0-9, or '-'")
}
```